opentracing-spring-cloud-starter-zipkin/src/main/java/io/opentracing/contrib/spring/cloud/starter/zipkin/ZipkinAutoConfiguration.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.opentracing.contrib.spring.cloud.starter.zipkin;
+
+import brave.Tracing;
+import brave.opentracing.BraveTracer;
+import brave.sampler.BoundarySampler;
+import brave.sampler.CountingSampler;
+import brave.sampler.Sampler;
+import java.util.Collections;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import zipkin2.Span;
+import zipkin2.reporter.AsyncReporter;
+import zipkin2.reporter.Reporter;
+import zipkin2.reporter.Sender;
+import zipkin2.reporter.okhttp3.OkHttpSender;
+
+@Configuration
+@ConditionalOnClass(brave.opentracing.BraveTracer.class)
+@ConditionalOnMissingBean(io.opentracing.Tracer.class)
+@ConditionalOnProperty(value = "opentracing.zipkin.enabled", havingValue = "true", matchIfMissing = true)
+@AutoConfigureAfter(name = "me.snowdrop.opentracing.tracer.jaeger.JaegerAutoConfiguration")
+@AutoConfigureBefore(name = "io.opentracing.contrib.spring.web.autoconfig.TracerAutoConfiguration")
+@EnableConfigurationProperties(ZipkinConfigurationProperties.class)
+public class ZipkinAutoConfiguration {
+
+  @Autowired(required = false)
+  private List<ZipkinTracerCustomizer> tracerCustomizers = Collections.emptyList();
+
+  @Value("${spring.application.name:unknown-spring-boot}")
+  private String serviceName;
+
+  @Bean
+  @ConditionalOnMissingBean
+  public io.opentracing.Tracer tracer(Reporter<Span> reporter, Sampler sampler) {
+
+    final Tracing.Builder builder = Tracing.newBuilder()
+        .sampler(sampler)
+        .localServiceName(serviceName)
+        .spanReporter(reporter);
+
+    tracerCustomizers.forEach(c -> c.customize(builder));
+
+    return BraveTracer.create(builder.build());
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnClass(OkHttpSender.class)
+  @ConditionalOnExpression("'${opentracing.zipkin.http-sender.url:null}' != 'null'")
```

Comment:
I think we should default  the url to `http://localhost:9411/` to make it easier to start with this module. If people want to disable it quickly they can use `opentracing.zipkin.enabled=false`.

Reply:
url should be http://localhost:9411/api/v2/spans

Reply:
Done